### PR TITLE
Move all the codes for simulator into new directory

### LINKED ISSUE #191: move simulator-server go module to another one directory
Currently, we have one go module in root dir.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/go.mod

And, we need to create other go modules for these KEPs. 
https://github.com/kubernetes-sigs/kube-scheduler-simulator/tree/master/keps/140-scenario-based-simulation
https://github.com/kubernetes-sigs/kube-scheduler-simulator/tree/master/keps/159-scheduler-simulator-operator
https://github.com/kubernetes-sigs/kube-scheduler-simulator/tree/master/keps/184-scheduler-simulation

I would propose to put only dirs for each go module (+ simulator-web) in root dir like:
```
├── simulator-server/
├── simulator-web/
├── scheduler-simulation/
├── simulator-operator/
├── README.md, etc (other docs can be placed in root dir)
```

/assign
/priority important-soon
/kind cleanup

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Create `/simulator` directory and move all simulator-related codes to `/simulator` dir.

#### Which issue(s) this PR fixes:

Fixes #191

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,53 +3,40 @@ tools:
...
 
 .PHONY: docker_build_server
 docker_build_server: 
-	docker build -t simulator-server .
+	docker build -t simulator-server ./simulator/
 
 .PHONY: docker_build_front
 docker_build_front: 
-	docker build -t simulator-frontend ./web/
+	docker build -t simulator-frontend ./simulator-web/
```
Comment (196Ikuchil) [inline Makefile:39]: `./simulator-web/`
Is this (maybe) renamed directory included in this PR?

**Review by everpeace — APPROVED** [review]
LGTM!

**Review by 196Ikuchil — COMMENTED** [review]
It would affect all PRs which currently submitted. 
But if we don't do it now, it will be much harder later!😇

File: Makefile (hunk 1 above)
Comment (sanposhiho) [reply Makefile:39]: あ, Nice catch. It's a mistake.

**Comment by sanposhiho** [comment]
> It would affect all PRs which currently submitted.

Yeah.... 😅 Hope the rebasing won't be so hard since it need to change file paths and import paths.

**Comment by sanposhiho** [comment]
@196Ikuchil Fixed

**Comment by 196Ikuchil** [comment]
Thank you, nice work.

/lgtm

**Pull request merged**